### PR TITLE
On reroute failure retry it with ignore bw

### PR DESCRIPTION
**Pull request by timofei-durakov**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy [hunk 1]
```diff
@@ -603,12 +610,26 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
         def switchToManipulate = topology.activeSwitches.find { !(it.dpId in involvedSwitches) }
         def blockData = switchHelper.knockoutSwitch(switchToManipulate, mgmtFlManager)
         def isSwitchActivated = false
+        wait(WAIT_OFFSET) {
+            def prevHistorySize = northbound.getFlowHistory(flow.flowId).size()
+            Wrappers.timedLoop(4) {
+                //history size should no longer change for both flows, all retries should give up
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy:616]: no 'both flows' here

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy [hunk 2]
```diff
@@ -603,12 +610,26 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
...
+                def newHistorySize = northbound.getFlowHistory(flow.flowId).size()
+                assert newHistorySize == prevHistorySize
+                withPool {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy:619]: withPool can be removed. It is a wrapper that allows parallel operations like 'eachParallel'

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy [hunk 3]
```diff
@@ -603,12 +610,26 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
...
+                    assert northbound.getFlowStatus(flow.flowId).status == FlowState.DOWN
+                }
+            sleep(500)
+            }
+        }
         switchHelper.reviveSwitch(switchToManipulate, blockData)
+        def expectedZeroReroutesTimestamp = System.currentTimeSeconds()
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy:626]: I think its safer to save the timestamp before calling `reviveSwitch`

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/UpdateFlowStatusAction.java [hunk 4]
```diff
@@ -48,6 +48,9 @@ protected void perform(State from, State to, Event event, FlowRerouteContext con
         FlowStatus resultStatus = persistenceManager.getTransactionManager().doInTransaction(() -> {
             Flow flow = getFlow(flowId, FetchStrategy.DIRECT_RELATIONS);
             FlowStatus flowStatus = flow.computeFlowStatus();
+            if (stateMachine.isIgnoreBandwidth() && flowStatus == FlowStatus.UP) {
+                flowStatus = FlowStatus.DEGRADED;
```
Comment (dpoltavets) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/UpdateFlowStatusAction.java:52]: In the current implementation, we have DEGRADED status only when we could not find the protected path. But in this case, this status is set for another reason. Therefore, the message `Couldn't find non overlapping protected path` in `status_info` will be incorrect.

Comment (timofei-durakov) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/UpdateFlowStatusAction.java:54]: good catch, haven't seen this before rebase, fixed

**Pull request merged**